generators/server/templates/src/main/java/package/service/UserService.java.ejs

```diff
@@ -847,12 +848,16 @@ public class UserService {
             // Here we can't cater for all the use-cases, see:
             // https://github.com/jhipster/generator-jhipster/issues/7866
             // for the en_US or en_** issue.
-            // Since JHipter only have one "en" language, we will handle it here, for other languages, please handle it accordingly.
+            // Since JHipter only has one "en" language, we will handle it here,
+            // for other languages, please handle it accordingly.
             String locale = (String) details.get("locale");
-            if(locale.startsWith("en_") || locale.startsWith("en-")) {
-                locale = "en";
+            if (locale.startsWith("en_") || locale.startsWith("en-")) {
+                locale = defaultLocale;
             }
             user.setLangKey(locale.toLowerCase());
+        } else {
+            // set langKey to default if not specified by IdP
+            user.setLangKey(defaultLocale);
```

Comment:
If translations are enabled, then, can we use default language key `nativeLanguage` to set this value?

Reply:
Good idea. Fixed!

Reply:
i think it will be better we use the following, what do you think ? @mraible 
```
            user.setLangKey("<%= nativeLanguage %>");
```